File: src/main/java/me/itzisonn_/meazy/parser/ast/statement/ImportStatement.java
package me.itzisonn_.meazy.parser.ast.statement;

import lombok.Getter;
import me.itzisonn_.meazy.instruction.InstructionsSet;
import me.itzisonn_.meazy.parser.ast.Statement;
import me.itzisonn_.meazy.runtime.environment.Environment;
import org.jspecify.annotations.NullMarked;

@Getter
@NullMarked
public class ImportStatement implements LocalStatement {
    private final String name;

    public ImportStatement(String name) {
        this.name = name;
    }

    @Override
    public void emit(InstructionsSet instructionsSet, Environment environment, Statement parent) {}

    @Override
    public boolean alwaysReturns() {
        return false;
    }
}

File: src/main/kotlin/me/itzisonn_/meazy/parser/ast/statement/ImportStatement.kt
package me.itzisonn_.meazy.parser.ast.statement

import me.itzisonn_.meazy.instruction.InstructionsSet
import me.itzisonn_.meazy.parser.ast.ParentMap
import me.itzisonn_.meazy.parser.ast.ProgramUnit
import me.itzisonn_.meazy.runtime.environment.Environment

class ImportStatement(val name: String) : LocalStatement {
    override val children = setOf<ProgramUnit>()

    context(parents: ParentMap)
    override fun emit(instructions: InstructionsSet, environment: Environment) {}

    override fun alwaysReturns() = false
}
